src/main/java/randoop/generation/OrienteeringSelection.java

```diff
@@ -0,0 +1,119 @@
+package randoop.generation;
+
+import java.util.HashMap;
+import java.util.Map;
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.util.Randomness;
+import randoop.util.SimpleList;
+
+/**
+ * Biases input selection towards sequences that have lower execution cost. Execution cost is
+ * measured by the number of method calls in a sequence and the time it takes to execute.
+ */
+public class OrienteeringSelection implements InputSequenceSelector {
+  /** Map from a sequence to its weight. */
+  private final Map<Sequence, Double> weightMap = new HashMap<>();
+
+  /** Map from a sequence to the number of times it was selected. */
+  private final Map<Sequence, Integer> sequenceSelectionCount = new HashMap<>();
+
+  /**
+   * Cache for the square root of the number of method calls in a sequence of statements. Once
+   * computed for a given sequence, the value is never updated.
+   */
+  private final Map<Sequence, Double> sequenceMethodSizeSqrt = new HashMap<>();
+
+  /**
+   * Map from a sequence to its approximate execution time in milliseconds. Once computed for a
+   * given sequence, the value is never updated.
+   */
+  private final Map<Sequence, Long> sequenceExecutionTime = new HashMap<>();
+
+  /**
+   * The sequence that was last selected by Orienteering. A reference is kept so that we can update
+   * its execution time once {@link ForwardGenerator} finishes executing the newly constructed
+   * sequence.
+   */
+  private Sequence lastSelectedSequence;
+
+  /**
+   * Bias input selection towards lower cost sequences.
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  @Override
+  public Sequence selectInputSequence(SimpleList<Sequence> candidates) {
+    updateWeightMapForCandidates(candidates);
+    lastSelectedSequence = Randomness.randomMemberWeighted(candidates, weightMap);
+    incrementCountInMap(sequenceSelectionCount, lastSelectedSequence);
+    return lastSelectedSequence;
+  }
+
+  /**
+   * Update the weights of the candidates in the given list
+   *
+   * @param candidates list of candidates
+   */
+  private void updateWeightMapForCandidates(SimpleList<Sequence> candidates) {
+    for (int i = 0; i < candidates.size(); i++) {
+      Sequence candidate = candidates.get(i);
+
+      // If we haven't computed the square root of the method size of this sequence,
+      // compute it and permanently store it.
+      Double methodSizeSqrt = sequenceMethodSizeSqrt.get(candidate);
+      if (methodSizeSqrt == null) {
+        methodSizeSqrt = Math.sqrt(candidate.numMethodCalls());
+        sequenceMethodSizeSqrt.put(candidate, methodSizeSqrt);
+      }
+
+      // If this sequence has not been selected yet, don't compute its weight and
+      // use its default value.
```

Comment:
Even if not computing its weight, I think it should be put in the weight map (or other data structure for managing a collection of weights).  That is more consistent, and it enables other documentation.

Reply:
I was not sure about this comment. However, to address it, I've changed it so that the weight map gets updated for a given sequence, regardless of whether or not the sequence has been selected yet or if the sequence doesn't have a computed execution time. In cases where selection count and execution time is not available for a candidate sequence, I map that sequence to it's default weight computed by sequence.getWeight() (instead of using a weight that would be obtained by the formula from GRT).